connectivity/drivers/wifi/esp8266-driver/ESP8266/ESP8266.h

```diff
@@ -455,9 +455,10 @@ class ESP8266 {
     /**
      * Stop board's and ESP8266's UART flow control
      *
+     * @param board_only    true to apply to board only, false to apply both
      * @return true if started
      */
-    bool stop_uart_hw_flow_ctrl();
+    bool stop_uart_hw_flow_ctrl(bool board_only);
```

Comment:
As this is public method, this would be breaking, wouldn't it?

What if we add overload adding boolean argument, we could deprecate the old method if it is necessary.

Reply:
How about C++ default argument:
```c++
bool stop_uart_hw_flow_ctrl(bool board_only = false);
```

The call to `stop_uart_hw_flow_ctrl()` without `board_only` specified will translate to `stop_uart_hw_flow_ctrl(false)` for compatibility.

Reply:
Also was considering this, https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rf-default-args - lets go with the default parameter

Reply:
Updated to go with default parameter